lib/vast/Frontend/GenAction.cpp

```diff
@@ -113,12 +124,14 @@ namespace vast::cc {
         void HandleInterestingDecl(clang::DeclGroupRef /* decl */) override {
             VAST_UNIMPLEMENTED;
         }
+        void emit_backend_output(clang::BackendAction backend_action,
+                                 owning_module_ref mlir_module, mcontext_t *mctx)
+        {
+            llvm::LLVMContext llvm_context;
+            target::llvmir::register_vast_to_llvm_ir(*mctx);
+            target::llvmir::prepare_hl_module(mlir_module.get());
```

Comment:
this has nothing with `hl` module prepare. Rename to something like `setup_lowering_pipeline`

Reply:
idealy we create single entity which will take `mlir_target` object and will setup required passes